File: frond/wu-ai-prdc-code/src/api/collaborationController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request'

/** 开始协作 POST /collaboration/start/${param0} */
export async function startCollaboration(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.startCollaborationParams,
  options?: { [key: string]: any }
) {
  const { appId: param0, ...queryParams } = params
  return request<API.BaseResponseLong>(`/collaboration/start/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  })
}

/** 添加协作者 POST /collaboration/add/${param0} */
export async function addCollaborator(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.addCollaboratorParams,
  body: API.CollaborationRequest,
  options?: { [key: string]: any }
) {
  const { collaborationId: param0, ...queryParams } = params
  return request<API.BaseResponseBoolean>(`/collaboration/add/${param0}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  })
}

/** 退出协作 POST /collaboration/exit/${param0} */
export async function exitCollaboration(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.exitCollaborationParams,
  options?: { [key: string]: any }
) {
  const { collaborationId: param0, ...queryParams } = params
  return request<API.BaseResponseBoolean>(`/collaboration/exit/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  })
}

/** 获取协作记录 GET /collaboration/record/${param0} */
export async function getCollaborationRecord(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getCollaborationRecordParams,
  options?: { [key: string]: any }
) {
  const { collaborationId: param0, ...queryParams } = params
  return request<API.BaseResponseCollaborationRecord>(`/collaboration/record/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  })
}

/** 根据应用ID获取协作记录 GET /collaboration/record/by-app/${param0} */
export async function getCollaborationRecordByAppId(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getCollaborationRecordByAppIdParams,
  options?: { [key: string]: any }
) {
  const { appId: param0, ...queryParams } = params
  return request<API.BaseResponseCollaborationRecord>(`/collaboration/record/by-app/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  })
}

/** 获取在线好友列表 GET /collaboration/online-friends */
export async function getOnlineFriends(
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseUserVO[]>(`/collaboration/online-friends`, {
    method: 'GET',
    ...(options || {}),
  })
}

/** 获取协作成员列表 GET /collaboration/members/${param0} */
export async function getCollaborationMembers(
  params: API.getCollaborationRecordParams,
  options?: { [key: string]: any }
) {
  const { collaborationId: param0, ...queryParams } = params
  return request<API.BaseResponseListCollaborationMember>(`/collaboration/members/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  })
}

File: backend/wu-ai-pdce/src/main/java/com/wuyinai/wuaipdce/controller/CollaborationController.java
package com.wuyinai.wuaipdce.controller;

import com.wuyinai.wuaipdce.common.BaseResponse;
import com.wuyinai.wuaipdce.common.ResultUtils;
import com.wuyinai.wuaipdce.model.dto.collaboration.CollaborationRequest;
import com.wuyinai.wuaipdce.model.entity.CollaborationRecord;
import com.wuyinai.wuaipdce.model.entity.User;
import com.wuyinai.wuaipdce.model.vo.CollaborationMemberVO;
import com.wuyinai.wuaipdce.service.CollaborationService;
import com.wuyinai.wuaipdce.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 协作 控制层。
 *
 * @author <a href="https://github.com/wuyinai">程序实习生奈奈</a>
 */
@RestController
@RequestMapping("/collaboration")
public class CollaborationController {

    @Resource
    private CollaborationService collaborationService;

    @Resource
    private UserService userService;

    /**
     * 开始协作
     * @param appId 应用ID
     * @param request 请求对象
     * @return 协作记录ID
     */
    @PostMapping("/start/{appId}")
    public BaseResponse<Long> startCollaboration(@PathVariable Long appId, HttpServletRequest request) {
        User loginUser = userService.getLoginUser(request);
        Long collaborationId = collaborationService.startCollaboration(appId, loginUser.getId());
        return ResultUtils.success(collaborationId);
    }

    /**
     * 添加协作者
     * @param collaborationId 协作记录ID
     * @param request 协作请求，包含要添加的用户ID
     * @return 成功响应
     */
    @PostMapping("/add/{collaborationId}")
    public BaseResponse<Boolean> addCollaborator(@PathVariable Long collaborationId, @RequestBody CollaborationRequest request) {
        collaborationService.addCollaborator(collaborationId, request.getUserId());
        return ResultUtils.success(true);
    }

    /**
     * 退出协作
     * @param collaborationId 协作记录ID
     * @return 成功响应
     */
    @PostMapping("/exit/{collaborationId}")
    public BaseResponse<Boolean> exitCollaboration(@PathVariable Long collaborationId) {
        collaborationService.exitCollaboration(collaborationId);
        return ResultUtils.success(true);
    }

    /**
     * 获取协作记录
     * @param collaborationId 协作记录ID
     * @return 协作记录
     */
    @GetMapping("/record/{collaborationId}")
    public BaseResponse<CollaborationRecord> getCollaborationRecord(@PathVariable Long collaborationId) {
        CollaborationRecord collaborationRecord = collaborationService.getCollaborationRecord(collaborationId);
        return ResultUtils.success(collaborationRecord);
    }

    /**
     * 根据应用ID获取协作记录
     * @param appId 应用ID
     * @return 协作记录
     */
    @GetMapping("/record/by-app/{appId}")
    public BaseResponse<CollaborationRecord> getCollaborationRecordByAppId(@PathVariable Long appId) {
        CollaborationRecord collaborationRecord = collaborationService.getCollaborationRecordByAppId(appId);
        return ResultUtils.success(collaborationRecord);
    }

    /**
     * 获取在线好友列表
     * @param request 请求对象
     * @return 在线好友列表
     */
    @GetMapping("/online-friends")
    public BaseResponse<List<User>> getOnlineFriends(HttpServletRequest request) {
        User loginUser = userService.getLoginUser(request);
        List<User> onlineFriends = collaborationService.getOnlineFriends(loginUser.getId());
        return ResultUtils.success(onlineFriends);
    }

    /**
     * 获取协作者列表
     * @param collaborationId 协作记录ID
     * @return 协作者列表
     */
    @GetMapping("/members/{collaborationId}")
    public BaseResponse<List<CollaborationMemberVO>> getCollaborationMembers(@PathVariable Long collaborationId) {
        List<CollaborationMemberVO> collaborationMembers = collaborationService.getCollaborationMembers(collaborationId);
        return ResultUtils.success(collaborationMembers);
    }

}